packages/plugin-degenfans-alfafrens/src/actions/stakingConsultantAction.ts

```diff
@@ -0,0 +1,133 @@
+} from "@moxie-protocol/core";
+
+import { stakingConsultantTemplate } from "../templates";
+import { Staking, StakingSchema } from "../types";
+import { FarcasterMetadata, ftaService, MoxieUser, TwitterMetadata } from "@moxie-protocol/moxie-agent-lib";
+import { getHelpText, getStakingOptions, getUserData, StakingRequest } from "../utils/degenfansApi";
+import { z } from 'zod';
+export const stakingConsultantAction: Action = {
+    name: "GET_ALFAFRENS_STAKING_RECOMENDATION",
+    similes: [
+        "VIEW_ALFAFRENS_STAKING_RECOMENDATION"
+    ],
+    description: "get recomendation for AlfaFrens stakings",
+    suppressInitialMessage: true,
+    validate: async () => true,
+    handler: async (
+        runtime: IAgentRuntime,
+        message: Memory,
+        state: State,
+        _options: { [key: string]: unknown },
+        callback: HandlerCallback
+    ) => {
+        try {
+            elizaLogger.log("Starting GET_ALFAFRENS_STAKING_RECOMENDATION handler...");
+
+            // Initialize or update state
+            if (!state) {
+                state = (await runtime.composeState(message)) as State;
+            } else {
+                state = await runtime.updateRecentMessageState(state);
+            }
+
+            const moxieUserInfo: MoxieUser = state.moxieUserInfo as MoxieUser;
+
+            const context = composeContext({
+                state,
+                template: stakingConsultantTemplate,
+            });
+
+            const transferDetails = await generateObject({
+                runtime,
+                context,
+                modelClass: ModelClass.SMALL,
+                schema: StakingSchema,
+            });
+            let {
+                userAddress,
+                amount,
+                mysubs,
+                mystake,
+                minsubs,
+            } = transferDetails.object as {
+                amount: number;
+                userAddress: string;
+                mysubs: boolean;
+                mystake: boolean;
+                minsubs: number;
+            };
+
+            const userData = getUserData(moxieUserInfo);
+
+            const stakingData: StakingRequest = { amount: amount, mysubs: mysubs, mystake: mystake, minsubs: minsubs };
+            const resp = await getStakingOptions(userData, userAddress, stakingData);
+            let tbl: string = "";
+            console.log(resp);
+            if (resp.status == 200) {
+                tbl += "\n";
+                if (resp.data && resp.data.result && resp.data.result.stakingOptions && resp.data.result.stakingOptions.length > 0) {
+                    tbl += "|rank|AlfaFrens Channel|ROI Spark/mo|current stake|\n";
+                    tbl += "|------:|:--------|----:|------|\n";
+                    resp.data.result.stakingOptions.forEach(e => {
+                        tbl += "|#" + e.rank + "|[" + e.name + "|https://alfafrens.com/channel/" + e.channelAddress + "]|" + e.roi + "|" + e.currentStake + "|\n";
```

Comment:
I see you wanna link the user with alfafrens link, best to do it this way:

```suggestion
                        tbl += "|#" + e.rank + "|[" + e.name + "](https://alfafrens.com/channel/" + e.channelAddress + ")|" + e.roi + "|" + e.currentStake + "|\n";
```

atm it's displaying like this, which makes it spillover to the next column:

<img width="961" alt="Screenshot 2025-03-27 at 23 46 17" src="https://github.com/user-attachments/assets/9930835b-68f7-4dc5-9769-1cff25626efe" />